### ISSUE TITLE
Wrong host in mail URLs

### ISSUE THREAD

**Issue by antillas21** [errbit/errbit#1101]
Hi,

Just deployed errbit to a droplet at digital ocean.

I followed the configuration instructions in your doc pages, and added these ENV vars to the `.env` file:

```env
ERRBIT_HOST=errbit.aantillon.com
ERRBIT_PROTOCOL=http
ERRBIT_EMAIL_FROM='errbit@aantillon.com'
EMAIL_DELIVERY_METHOD=:smtp
SMTP_SERVER=smtp.sendgrid.net
SMTP_PORT=587
SMTP_AUTHENTICATION=:plain
SMTP_USERNAME=sendgrid_username_here
SMTP_PASSWORD=sendgrid_password_here
SMTP_DOMAIN=errbit.aantillon.com
```

then checked in the production server if `ActionMailer` is properly configured:

```ruby
irb(main):001:0> Errbit::Config.host
=> "errbit.aantillon.com"

irb(main):002:0> Errbit::Config.email_from
=> "errbit@aantillon.com"

irb(main):003:0> ActionMailer::Base.smtp_settings
=> {:address=>"smtp.sendgrid.net", :port=>587, :authentication=>:plain, :user_name=>"sengrid_username_here", :password=>"sendgrid_password_here", :domain=>"errbit.aantillon.com"}

irb(main):004:0> ActionMailer::Base.default_url_options
=> {:host=>"errbit.aantillon.com", :protocol=>"http"}
```

All seems to be OK... but when, for example, requesting to reset a user password, the email received displays the following:

```
from: errbit@example.com
http://errbit.example.com/users/password/edit?reset_password_token=resetPasswordTokenHere

and the link to the errbit instance: http://errbit.example.com/
```

for some reason, unknown to me, it falls back to the default ENV vars?

Could anyone provide a hint to what I'm missing here?

**Comment by stevecrozz** [comment]
This all seems fine. I can see that the configuration looks good and it
looks like the configuration is effective in your console session. The only
thing that comes to mind is that maybe the configuration is not effective
in the server process.

To debug it, I'd consider adding some logging to the application to make
sure all this info that you've already double-checked in the console
(especially ActionMailer::Base.smtp_settings and
ActionMailer::Base.default_url_options) look good also in the actual web
process. I don't have any idea why that would be the case, but it seems
like that's what's happening.

Maybe log those things at the bottom
of config/initializers/action_mailer.rb using Rails.logger.info or
something similar, restart the rails process and verify these settings are
correct in that context.

This is extra confusing because it looks like some of your configuration is
taking effect because you said you're receiving emails and that means the
email configuration is mostly working.

Another way to go about this would be to bring up your console again where
the configuration is correct and send a password recovery email from there,
like this and if that comes through correctly, then you'll know the
configuration itself is working and that something is wrong with the web
process specifically:
user = User.find('some-id')
user.send_reset_password_instructions

**Comment by antillas21** [comment]
Thanks for the suggestions @stevecrozz ... I retrieved my user from the console and called `.send_reset_password_instructions` on it, the email I received displays all the correct information

```
from: errbit@aantillon.com
reset password link: http://errbit.aantillon.com/users/password/edit?reset_password_token=resetPasswordToken
errbit instance URL: http://errbit.aantillon.com/
```

but still, when going through the app in the browser, I get the default settings on all emails.

I'm thinking if this could be caused by the fact that I first deployed the app without customizing ENV vars, then I created an nginx vhost file, after that I added the correct ENV vars. I do recall restarting all processes (`puma, nginx`), and even performing a new deployment.
